Adds "Best practices" section. First topic is management of Keyboard … (#12)
Coppernic/coppernic.github.io: docs/bestpractices/keyboard_peripherals.md

---
sidebar_position: 1
title: Keyboard peripherals
---

Keyboard peripherals
==================================

About keyboard Peripherals
------------------------

Some peripherals available on Coppernic devices are managed as keyboard-type device by Android OS.
It concerns some barcode reader, RFID reader, MRZ reader.

> A device like [HID reader on Access-ER](/docs/product/access-er/hid) can act as CCID device, or as a keyboard wedge, depending its configuration.

Reviewer: depending **on**

Author: bien vu merci ! Sur ce genre de correction d'orthographe, n'hésite pas à utiliser le bouton "Add suggestion" ca permet de corriger rapidement sans faire de commit en local

Impact on Android activity life cycle
------------------------

By default an Android Activity treats a keyboard plug as a configuration change.

https://developer.android.com/guide/topics/resources/runtime-changes#HandlingTheChange

> This leads to Activity being recreated when a keyboard peripheral is power on or powered off, and can seem to a bad user experience, or seems there is a bug.

Reviewer 2: Il manque un mot sur cette ligne non ? "and can seem to a bad user experience"

Author: Tu as raison, j'ai changé la tournure de la phrase

> This leads to Activity being recreated when a keyboard peripheral is power on or powered off, and can seem to a bad user experience, or seems there is a bug.

Author: ```suggestion
This leads to Activity being recreated when a keyboard peripheral is power on or powered off, and may seem to be a bad user experience, or look like a bug.
```
